gemma-core/src/test/java/ubic/gemma/core/search/source/HibernateSearchSourceTest.java

```diff
@@ -0,0 +1,54 @@
+package ubic.gemma.core.search.source;
+
+import org.apache.lucene.search.Query;
+import org.hibernate.search.FullTextSession;
+import org.hibernate.search.Search;
+import org.hibernate.search.query.dsl.QueryBuilder;
+import org.junit.Test;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.test.context.ContextConfiguration;
+import ubic.gemma.core.util.test.BaseDatabaseTest;
+import ubic.gemma.model.expression.experiment.ExpressionExperiment;
+import ubic.gemma.model.genome.Taxon;
+import ubic.gemma.persistence.util.TestComponent;
+
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@ContextConfiguration
+public class HibernateSearchSourceTest extends BaseDatabaseTest {
+
+    @Configuration
+    @TestComponent
+    static class HibernateSearchSourceContextConfiguration extends BaseDatabaseTestContextConfiguration {
+    }
+
+    @Test
+    public void testSearch() {
+        FullTextSession fts = Search.getFullTextSession( sessionFactory.getCurrentSession() );
+        QueryBuilder queryBuilder = fts.getSearchFactory().buildQueryBuilder().forEntity( ExpressionExperiment.class ).get();
+        Query query = queryBuilder.keyword().onField( "shortName" ).matching( "hello" )
+                .createQuery();
+
+        //noinspection unchecked
+        List<ExpressionExperiment> results = Search.getFullTextSession( Search.getFullTextSession( sessionFactory.getCurrentSession() ) )
+                .createFullTextQuery( query, ExpressionExperiment.class )
+                .list();
+        assertThat( results ).isEmpty();
+
+        Taxon taxon = new Taxon();
+        fts.persist( taxon );
+        ExpressionExperiment ee = new ExpressionExperiment();
+        ee.setShortName( "hello" );
+        ee.setTaxon( taxon );
+        fts.persist( ee );
+        fts.flushToIndexes();
```

Comment:
In practice, this would be a simple flush(), would that work?

Reply:
Ok, so what happens is that Hibernate Search update the index after a sufficient large batch of changes has been done. This method triggers an immediate index update.